src/core/utils/searchConversationHistory.ts

```diff
@@ -0,0 +1,33 @@
+import { ConversationHistory } from '../context/types';
+
+interface SearchResult {
+  id: string;
+  title: string;
+}
+
+/**
+ * Searches through conversation history for matches in titles
+ * @param searchTerm - The term to search for
+ * @param conversations - The conversation store object
+ * @returns Array of matching conversation IDs and titles
+ */
+export function searchConversationHistory(
+  searchTerm: string,
+  conversations: Record<string, ConversationHistory>,
+): SearchResult[] {
```

Comment:
I can envision this function being run on input change (maybe after input.length > 3 so that we don't get immediate matches for the first letter?) as a user types to search for conversation title